### DISCUSSION TITLE
How to configure a global navigation guard?

### DISCUSSION THREAD

**Discussion by huyixi** [posva/unplugin-vue-router#475] [Questions and Help] [answered]
When a user navigates to a page, I need to check their permissions and render the page accordingly based on those permissions. How can I achieve this with a global router guard?

**Comment by PindaPixel** [comment]
There's a ton of ways to do this, and it doesn't necessarily have to be a global navigation guard. For example, you could keep track of the user and their permissions in a global store and render the page based on the state of that store.

From what I can tell, navigation guards are more meant to redirect a user to a different page. Alternatively, you alter store state from a navigation guard: https://router.vuejs.org/guide/advanced/navigation-guards.html#Global-injections-within-guards

**Reply by posva** [reply]
FYI In most scenarios, the guard will be the simplest solution to avoid ending up in a page the user hasn't access and where things just break (e.g. failing permissions)

**Comment by posva** [comment, marked as answer]
You can attach guards with `router.beforeEach()` and [other guards](https://router.vuejs.org/guide/advanced/navigation-guards.html#Global-Before-Guards) like usual, after creating your router instance.
